code/game/machinery/vending.dm

```diff
@@ -602,12 +606,21 @@
 		flick(src.icon_vend,src)
 	spawn(src.vend_delay)
 		playsound(src.loc, 'sound/machines/vending.ogg', 35, 1)
-		new R.product_path(get_turf(src))
+		var/obj/vended = new R.product_path(get_turf(src))
 		src.status_message = ""
 		src.status_error = 0
 		src.vend_ready = 1
 		currently_vending = null
 		SSnanoui.update_uis(src)
+		if(istype(vended,/obj/item/weapon/reagent_containers/))
+			var/obj/item/weapon/reagent_containers/RC = vended
+			if(RC.reagents)
+				if(heating)
+					use_power(RC.reagents.set_temperature(heating_temperature))
+					cooling = 0 //Safety

```

Comment:
This doesn't make sense. Stop cooling or heating when something's vended...?

Reply:
It's a safety var change in case something weird happens, such as both heating and cooling being set to 1. Will be changed when the two vars are togther.